Assets/Scripts/Editor/Compilation/CodeRewriting/Partials/PartialTreeProcessor.cs

```diff
@@ -30,11 +30,12 @@ internal static SyntaxTree ProcessPartialTree(SyntaxTree tree, IEnumerable<strin
                 return tree;
             }
 
-            var combinedTypes = new Dictionary<string, List<TypeDeclarationSyntax>>();
-            var combinedUsingDirectives = new HashSet<UsingDirectiveSyntax>();
-            var combinedTypesDefined = new HashSet<string>();
+            // root key is namespace, inner key is type
+            var combinedTypes = new Dictionary<string, Dictionary<string, List<TypeDeclarationSyntax>>>();
```

Comment:
would be good to use named type - this way comment is not necessary and whoever works with the structure knows what it represents

Reply:
So how would you like this type to look like? Anyway, feel free to make it a type ;)

Reply:
no probs - I'll try